boot-microservices/src/main/java/org/ow2/proactive/boot/microservices/iam/util/JWTValidator.java

```diff
@@ -132,15 +109,15 @@ public Assertion validate(String jsonWebToken, String service) throws TicketVali
             // Extract JWT principal from attributes
             AttributePrincipal principal = new AttributePrincipalImpl(claims.getSubject());
 
-            //Create authentication assertion
+            // Create authentication assertion
             Assertion assertion = new AssertionImpl(new AttributePrincipalImpl(principal.getName(),
                                                                                claims.getClaimsMap()));
 
             logger.debug("Assertion generated for principal " + principal.getName());
 
             return assertion;
         } catch (MalformedClaimException e) {
-            throw new IAMException("");
+            throw new IAMException("Malformed Json Web Token acquired from IAM [" + jsonWebToken + "]");
```

Comment:
Can you link the exception cause `e` here?

Reply:
done